src/test/java/com/example/books/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,184 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+    private static final Long DEFAULT_ID = 1L;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void setup(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/insert-testing-user.sql",
+            "classpath:database/shoppingcart/insert-testing-shopping-cart.sql",
+            "classpath:database/books/insert-testing-book.sql",
+            "classpath:database/shoppingcart/insert-cart-item.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/remove-cart-item.sql",
+            "classpath:database/shoppingcart/remove-testing-shopping-cart.sql",
+            "classpath:database/books/remove-testing-book.sql",
+            "classpath:database/shoppingcart/remove-testing-user.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getShoppingCart_Ok() throws Exception {
+        MvcResult result = mockMvc.perform(get("/api/cart"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), ShoppingCartDto.class);
+        Assertions.assertNotNull(actual);
+
+        EqualsBuilder.reflectionEquals(
+                getShoppingCartDto(),
+                actual
+        );
+    }
+
+    @Test
+    @WithMockUser
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/insert-testing-user.sql",
+            "classpath:database/shoppingcart/insert-testing-shopping-cart.sql",
+            "classpath:database/books/insert-testing-book.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/shoppingcart/remove-testing-shopping-cart.sql",
+            "classpath:database/books/remove-testing-book.sql",
+            "classpath:database/shoppingcart/remove-testing-user.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)

```

Comment:
Please check if it's possible to move majority of scripts to class level to make methods annotations shorter